static/src/javascripts/projects/common/modules/commercial/creatives/expandable-v3.js

```diff
@@ -49,80 +51,79 @@ define([
                     });
-                }.bind(this));
+                });
                 break;
             case 'fixed':
-                fastdom.read(function () {
-                    this.scrollAmount = (window.pageYOffset - this.$adSlot.offset().top) + 'px';
-                }.bind(this));
+                this.scrollAmount = (scrollY - adSlotTop) + 'px';
                 fastdom.write(function () {
-                    $('.ad-exp--expand-scrolling-bg', $(this.$adSlot)).css('background-position', '50%' + this.scrollAmount);
-                }.bind(this));
+                    $('.ad-exp--expand-scrolling-bg', $(that.$adSlot)).css('background-position', '50%' + that.scrollAmount);
+                });
                 break;
             case 'fixed matching fluid250':
                 fastdom.write(function () {
-                    $('.ad-exp--expand-scrolling-bg', $(this.$adSlot)).addClass('ad-exp--expand-scrolling-bg-fixed');
-                }.bind(this));
+                    $('.ad-exp--expand-scrolling-bg', $(that.$adSlot)).addClass('ad-exp--expand-scrolling-bg-fixed');
+                });
                 break;
             case 'parallax':
-                fastdom.read(function () {
-                    this.scrollAmount = Math.ceil((window.pageYOffset - this.$adSlot.offset().top) * 0.3 * -1) + 20;
-                    this.scrollAmountP = this.scrollAmount + '%';
-                }.bind(this));
+                this.scrollAmount = Math.ceil((scrollY - adSlotTop) * 0.3 * -1) + 20;
+                this.scrollAmountP = this.scrollAmount + '%';
                 fastdom.write(function () {
-                    $('.ad-exp--expand-scrolling-bg', $(this.$adSlot)).addClass('ad-exp--expand-scrolling-bg-parallax');
-                    $('.ad-exp--expand-scrolling-bg', $(this.$adSlot)).css('background-position', '50%' + this.scrollAmountP);
-                }.bind(this));
+                    $('.ad-exp--expand-scrolling-bg', $(that.$adSlot)).addClass('ad-exp--expand-scrolling-bg-parallax');
+                    $('.ad-exp--expand-scrolling-bg', $(that.$adSlot)).css('background-position', '50%' + that.scrollAmountP);
+                });
                 break;
         }
     };
 
     ExpandableV3.prototype.listener = function () {
-        fastdom.read(function () {
-            if (!this.initialExpandCounter && (window.pageYOffset + bonzo.viewport().height) > (this.$adSlot.offset().top + this.openedHeight)) {
+        var that = this;
+        if (!this.initialExpandCounter && (window.pageYOffset + bonzo.viewport().height) > (this.$adSlot.offset().top + this.openedHeight)) {
+            if (!storage.local.get('gu.commercial.expandable.' + this.params.ecid)) {
+                // expires in 1 week
+                var week = 1000 * 60 * 60 * 24 * 7;
+                storage.local.set('gu.commercial.expandable.' + this.params.ecid, true, { expires: Date.now() + week });
```

Comment:
I kind prefer when `storage.get / set` is inside fastdom. Those methods can be slow and throw exceptions. Now they'll have an impact on the other listeners of `throttledScroll`

Or maybe `ExpandableV3.listener` is inside a debounce and it makes no difference. Dunno, just raising the point